src/inet/AsyncDNSResolverSockets.cpp

```diff
@@ -172,8 +195,7 @@ INET_ERROR AsyncDNSResolverSockets::EnqueueRequest(DNSResolver &resolver)
 {
     INET_ERROR err = INET_NO_ERROR;
     int pthreadErr;
-
-    AsyncMutexLock();
+    LockHolder asyncMutexLock(*this);
```

Comment:
Similar to above, I think this pattern makes it easy to miss the acquisition of the lock.  I suggest always using an independent block to clearly demark the scope of the lock, with the lock being the only variable declared at the start of the block.  E.g.:

```
INET_ERROR AsyncDNSResolverSockets::EnqueueRequest(DNSResolver &resolver)
{
    INET_ERROR err = INET_NO_ERROR;
    int pthreadErr;

    {
        auto lock = LockHolder::AquireLock(*this);

        ...
    }

    return err;
}
```